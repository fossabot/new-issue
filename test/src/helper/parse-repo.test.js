/* eslint-env node, jest */

import {
  parseRepo,
} from '../../../src/helper/parse-repo'

describe('parse repo from url pathname', () => {
  it('should return null if pathname is missing or malformed', () => {
    expect(parseRepo()).toBeNull()
    expect(parseRepo('')).toBeNull()
    expect(parseRepo({})).toBeNull()
    expect(parseRepo(null)).toBeNull()
    expect(parseRepo('bad')).toBeNull()
    expect(parseRepo('/bad')).toBeNull()
  })

  it('should parse /owner/repo correctly', () => {
    expect(parseRepo('/name/repo')).toMatchSnapshot()
    expect(parseRepo('/1/2')).toMatchSnapshot()
    expect(parseRepo('/1.1/2.2')).toMatchSnapshot()
    expect(parseRepo('/n__+/-__r')).toMatchSnapshot()
  })

  it('should parse /owner/repo#hash correctly', () => {
    expect(parseRepo('/name/repo')).toMatchSnapshot()
    expect(parseRepo('/1/2')).toMatchSnapshot()
    expect(parseRepo('/1.1/2.2')).toMatchSnapshot()
    expect(parseRepo('/n__+/-__r')).toMatchSnapshot()
  })

  it('should parse /owner/repo/type correctly', () => {
    expect(parseRepo('/name/repo/type')).toMatchSnapshot()
    expect(parseRepo('/1/2/3')).toMatchSnapshot()
    expect(parseRepo('/1.1/2.2/3.3')).toMatchSnapshot()
    expect(parseRepo('/n__+/-__r/_---_')).toMatchSnapshot()
  })

  it('should parse /owner/repo/type#hash correctly', () => {
    expect(parseRepo('/name/repo/type', '#1a2b3c')).toMatchSnapshot()
    expect(parseRepo('/1/2/3', '1a2b3c')).toMatchSnapshot()
    expect(parseRepo('/1.1/2.2/3.3', '#1a2b3c')).toMatchSnapshot()
    expect(parseRepo('/n__+/-__r/_---_', '1a2b3c')).toMatchSnapshot()
  })
})
