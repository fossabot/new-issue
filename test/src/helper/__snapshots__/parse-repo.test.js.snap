// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse repo from url pathname should parse /owner/repo correctly 1`] = `
Object {
  "configPaths": Array [
    "/name/repo/master/.new-issue",
    "/name/repo/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/name/index.json",
    "/wei/new-issue/master/repos/name/repo/index.json",
  ],
  "hash": undefined,
  "owner": "name",
  "repo": "repo",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo correctly 2`] = `
Object {
  "configPaths": Array [
    "/1/2/master/.new-issue",
    "/1/2/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/1/index.json",
    "/wei/new-issue/master/repos/1/2/index.json",
  ],
  "hash": undefined,
  "owner": "1",
  "repo": "2",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo correctly 3`] = `
Object {
  "configPaths": Array [
    "/1.1/2.2/master/.new-issue",
    "/1.1/2.2/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/1.1/index.json",
    "/wei/new-issue/master/repos/1.1/2.2/index.json",
  ],
  "hash": undefined,
  "owner": "1.1",
  "repo": "2.2",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo correctly 4`] = `
Object {
  "configPaths": Array [
    "/n__+/-__r/master/.new-issue",
    "/n__+/-__r/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/n__+/index.json",
    "/wei/new-issue/master/repos/n__+/-__r/index.json",
  ],
  "hash": undefined,
  "owner": "n__+",
  "repo": "-__r",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo#hash correctly 1`] = `
Object {
  "configPaths": Array [
    "/name/repo/master/.new-issue",
    "/name/repo/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/name/index.json",
    "/wei/new-issue/master/repos/name/repo/index.json",
  ],
  "hash": undefined,
  "owner": "name",
  "repo": "repo",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo#hash correctly 2`] = `
Object {
  "configPaths": Array [
    "/1/2/master/.new-issue",
    "/1/2/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/1/index.json",
    "/wei/new-issue/master/repos/1/2/index.json",
  ],
  "hash": undefined,
  "owner": "1",
  "repo": "2",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo#hash correctly 3`] = `
Object {
  "configPaths": Array [
    "/1.1/2.2/master/.new-issue",
    "/1.1/2.2/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/1.1/index.json",
    "/wei/new-issue/master/repos/1.1/2.2/index.json",
  ],
  "hash": undefined,
  "owner": "1.1",
  "repo": "2.2",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo#hash correctly 4`] = `
Object {
  "configPaths": Array [
    "/n__+/-__r/master/.new-issue",
    "/n__+/-__r/master/.new-issue/index.json",
    "/wei/new-issue/master/repos/n__+/index.json",
    "/wei/new-issue/master/repos/n__+/-__r/index.json",
  ],
  "hash": undefined,
  "owner": "n__+",
  "repo": "-__r",
  "type": undefined,
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type correctly 1`] = `
Object {
  "configPaths": Array [
    "/name/repo/master/.new-issue",
    "/name/repo/master/.new-issue/type.json",
    "/wei/new-issue/master/repos/name/index.json",
    "/wei/new-issue/master/repos/name/repo/type.json",
  ],
  "hash": undefined,
  "owner": "name",
  "repo": "repo",
  "type": "type",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type correctly 2`] = `
Object {
  "configPaths": Array [
    "/1/2/master/.new-issue",
    "/1/2/master/.new-issue/3.json",
    "/wei/new-issue/master/repos/1/index.json",
    "/wei/new-issue/master/repos/1/2/3.json",
  ],
  "hash": undefined,
  "owner": "1",
  "repo": "2",
  "type": "3",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type correctly 3`] = `
Object {
  "configPaths": Array [
    "/1.1/2.2/master/.new-issue",
    "/1.1/2.2/master/.new-issue/3.3.json",
    "/wei/new-issue/master/repos/1.1/index.json",
    "/wei/new-issue/master/repos/1.1/2.2/3.3.json",
  ],
  "hash": undefined,
  "owner": "1.1",
  "repo": "2.2",
  "type": "3.3",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type correctly 4`] = `
Object {
  "configPaths": Array [
    "/n__+/-__r/master/.new-issue",
    "/n__+/-__r/master/.new-issue/_---_.json",
    "/wei/new-issue/master/repos/n__+/index.json",
    "/wei/new-issue/master/repos/n__+/-__r/_---_.json",
  ],
  "hash": undefined,
  "owner": "n__+",
  "repo": "-__r",
  "type": "_---_",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type#hash correctly 1`] = `
Object {
  "configPaths": Array [
    "/name/repo/master/.new-issue",
    "/name/repo/master/.new-issue/type.json",
    "/wei/new-issue/master/repos/name/index.json",
    "/wei/new-issue/master/repos/name/repo/type.json",
    "/name/repo/1a2b3c/.new-issue",
    "/name/repo/1a2b3c/.new-issue/type.json",
    "/wei/new-issue/1a2b3c/repos/name/index.json",
    "/wei/new-issue/1a2b3c/repos/name/repo/type.json",
  ],
  "hash": "1a2b3c",
  "owner": "name",
  "repo": "repo",
  "type": "type",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type#hash correctly 2`] = `
Object {
  "configPaths": Array [
    "/1/2/master/.new-issue",
    "/1/2/master/.new-issue/3.json",
    "/wei/new-issue/master/repos/1/index.json",
    "/wei/new-issue/master/repos/1/2/3.json",
    "/1/2/1a2b3c/.new-issue",
    "/1/2/1a2b3c/.new-issue/3.json",
    "/wei/new-issue/1a2b3c/repos/1/index.json",
    "/wei/new-issue/1a2b3c/repos/1/2/3.json",
  ],
  "hash": "1a2b3c",
  "owner": "1",
  "repo": "2",
  "type": "3",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type#hash correctly 3`] = `
Object {
  "configPaths": Array [
    "/1.1/2.2/master/.new-issue",
    "/1.1/2.2/master/.new-issue/3.3.json",
    "/wei/new-issue/master/repos/1.1/index.json",
    "/wei/new-issue/master/repos/1.1/2.2/3.3.json",
    "/1.1/2.2/1a2b3c/.new-issue",
    "/1.1/2.2/1a2b3c/.new-issue/3.3.json",
    "/wei/new-issue/1a2b3c/repos/1.1/index.json",
    "/wei/new-issue/1a2b3c/repos/1.1/2.2/3.3.json",
  ],
  "hash": "1a2b3c",
  "owner": "1.1",
  "repo": "2.2",
  "type": "3.3",
}
`;

exports[`parse repo from url pathname should parse /owner/repo/type#hash correctly 4`] = `
Object {
  "configPaths": Array [
    "/n__+/-__r/master/.new-issue",
    "/n__+/-__r/master/.new-issue/_---_.json",
    "/wei/new-issue/master/repos/n__+/index.json",
    "/wei/new-issue/master/repos/n__+/-__r/_---_.json",
    "/n__+/-__r/1a2b3c/.new-issue",
    "/n__+/-__r/1a2b3c/.new-issue/_---_.json",
    "/wei/new-issue/1a2b3c/repos/n__+/index.json",
    "/wei/new-issue/1a2b3c/repos/n__+/-__r/_---_.json",
  ],
  "hash": "1a2b3c",
  "owner": "n__+",
  "repo": "-__r",
  "type": "_---_",
}
`;
